apps/ui/components/application/Page2.tsx

```diff
@@ -214,15 +214,23 @@ const getApplicationEventsWhichMinDurationsIsNotFulfilled = (
 
 const Page2 = ({ application, onNext }: Props): JSX.Element => {
   const { t } = useTranslation();
-
+  const [selectedReservationUnit, setSelectedReservationUnit] =
+    useState<number>(0);
   const [successMsg, setSuccessMsg] = useState("");
   const [errorMsg, setErrorMsg] = useState("");
   const [minDurationMsg, setMinDurationMsg] = useState(true);
   const router = useRouter();
   const openingHours = filterNonNullable(
-    application?.applicationEvents?.[0]?.eventReservationUnits?.[0]
-      ?.reservationUnit.applicationRoundTimeSlots
+    application?.applicationEvents?.[0]?.eventReservationUnits?.[
+      selectedReservationUnit
+    ]?.reservationUnit.applicationRoundTimeSlots
+  );
+  const reservationUnitOptions = filterNonNullable(
+    application?.applicationEvents?.[0]?.eventReservationUnits?.map(
+      (n, idx) => ({ value: idx, label: n?.reservationUnit.nameFi ?? "" })
```

Comment:
I'd use `pk` as value for the selected reservationUnit.
Requires the use of find to map the value, but it's not brittle to cache reset / refetch, and React keys are correct (in case it would dynamically be removed from the array).

Reply:
Yep, a more robust way to do it. Refactored.